arangod/Aql/ExecutionPlan.cpp

```diff
@@ -526,8 +526,16 @@ bool ExecutionPlan::contains(ExecutionNode::NodeType type) const {
 }
 
 /// @brief increase the node counter for the type
-void ExecutionPlan::increaseCounter(ExecutionNode::NodeType type) noexcept {
+void ExecutionPlan::increaseCounter(ExecutionNode const& node) noexcept {
+  auto const type = node.getType();
   ++_typeCounts[type];
+  if (type == ExecutionNode::ENUMERATE_COLLECTION) {
```

Comment:
I guess nodes of type `INDEX` are not considered here because they all start up as nodes of type ENUMERATE_COLLECTION initially, and are only converted to INDEX nodes by the optimizer, which already considers index hints in the `use-indexes` optimizer rule.
If so, adding a code comment with the explanation would be nice. Thanks!

Reply:
Yes you are right. Will add a comment